### DISCUSSION TITLE
Retrain with domain specific dataset

### DISCUSSION THREAD

**Discussion by erlakshmi123** [s-JoL/Open-Llama#50] [General]
Hi , Using s-JoL/Open-Llama-V1-pretrain. Can I merge my own vocab and retrain the unsupervised part?. After that I want to do instrtuct tuning. Could you suggest steps for that?

**Comment by s-JoL** [comment]
Of course, you can find in the utils section on how to train a tokenizer model and merge it with an existing tokenizer model. Additionally, you can learn how to convert and add new token embeddings to an existing model checkpoint. After completing these steps, you just need to modify the pretrain_config and specify the path of your custom checkpoint in train-ckpt.

**Reply by erlakshmi123** [reply]
thank you. Follow up when i do that should i still pass the pile/wudao datasets? or only my new dataset?. Also the v2_pretrained model in hugguingface is it a checkpoint with both pile and wudao dataset?
data:
  mode: "pretrain"
  data: 
    wudao: "data/pretrain_data/part-wudao*.jsonl.zst"
    the_pile: "data/pretrain_data/part-pile-1*.jsonl.zst"
  pad_to_max: False

**Reply by s-JoL** [reply]
The v2_pretrained model is pretrained using both the pile and wudao dataset, so you only need to fill in your own dataset.

**Reply by erlakshmi123** [reply]
Do I need to download pile and wudao datasets or just update my dataset from the checkpoint?. Should i modify train_lm to only include my dataset?

**Reply by s-JoL** [reply]
you just need to replace the pile to your own data in the config.

**Reply by erlakshmi123** [reply]
thanks. Also where do i get this data data/llama_raw_ckpt/7B/consolidated.00.pth

**Reply by s-JoL** [reply]
this is the path to the checkpoint, use s-JoL/Open-Llama-V2-pretrain as the path is alright.

**Reply by erlakshmi123** [reply]
one final question.. since the model is trained on both pile and wudao dataset. In merge_tokenizer i will use configs/tokenizer_models/llama_tokenizer_extended.model as the base raw model ?

**Reply by erlakshmi123** [reply]
also assuming because convert_ckpt is saving in torch.save(ckpt, "data/llama_raw_ckpt/7B/extended.pth"). traim_lm should load from here?

**Reply by s-JoL** [reply]
yes to both two questions.
